File: app/src/main/java/com/example/beerhive/beerdetail/Beer.java
package com.example.beerhive.beerdetail;

import android.os.Parcel;
import android.os.Parcelable;

public class Beer implements Parcelable {

    private String beerName;
    private String beerDescription;
    private String beerBrewerTips;
    private String beerImageUrl;

    String getBeerName() {
        return beerName;
    }

    public void setBeerName(String beerName) {
        this.beerName = beerName;
    }

    public String getBeerDescription() {
        return beerDescription;
    }

    public void setBeerDescription(String beerDescription) {
        this.beerDescription = beerDescription;
    }

    public String getBeerBrewerTips() {
        return beerBrewerTips;
    }

    public void setBeerBrewerTips(String beerBrewerTips) {
        this.beerBrewerTips = beerBrewerTips;
    }

    public String getBeerImageUrl() {
        return beerImageUrl;
    }

    public void setBeerImageUrl(String beerImageUrl) {
        this.beerImageUrl = beerImageUrl;
    }

    public static Parcelable.Creator<Beer> getCREATOR() {
        return CREATOR;
    }

    public Beer(){

    }
    protected Beer(Parcel in) {
        beerName = in.readString();
        beerDescription = in.readString();
        beerBrewerTips = in.readString();
        beerImageUrl = in.readString();
    }

    public static final Parcelable.Creator<Beer> CREATOR = new Parcelable.Creator<Beer>() {
        @Override
        public Beer createFromParcel(Parcel in) {
            return new Beer(in);
        }

        @Override
        public Beer[] newArray(int size) {
            return new Beer[size];
        }
    };

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(beerName);
        dest.writeString(beerDescription);
        dest.writeString(beerBrewerTips);
        dest.writeString(beerImageUrl);
    }
}

File: app/src/main/java/com/example/beerhive/beerdetail/Beer.kt
package com.example.beerhive.beerdetail

import android.os.Parcel
import android.os.Parcelable

data class Beer(var beerName: String? = null,
                var beerDescription: String? = null,
                var beerBrewerTips: String? = null,
                var beerImageUrl: String? = null):Parcelable {
    constructor(parcel: Parcel) : this(
            parcel.readString(),
            parcel.readString(),
            parcel.readString(),
            parcel.readString())

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(beerName)
        parcel.writeString(beerDescription)
        parcel.writeString(beerBrewerTips)
        parcel.writeString(beerImageUrl)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<Beer> {
        override fun createFromParcel(parcel: Parcel): Beer {
            return Beer(parcel)
        }

        override fun newArray(size: Int): Array<Beer?> {
            return arrayOfNulls(size)
        }
    }
}